### PR TITLE
chore: add mixed protocol feature gates

### PR DESCRIPTION
**Pull request by nilo19**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

We need mixed protocol feature gate to test mixed protocol lb typed service in cloud provider azure e2e test.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
chore: add mixed protocol feature gates
```

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
@nilo19 are you trying to add this to all user templates using external cloud provider as well or just e2e test templates?

We use kustomize to generate templates and this is where the kustomization file is defined: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/templates/test/ci/prow-external-cloud-provider/kustomization.yaml so you'll need to either 1) update the external flavor (if you want to change to be applied to the release external template example or 2) add a patch to add this for the test template

Also please add a description on the PR of what this does and why it's needed.

File: templates/test/ci/patches/apiserver.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+kind: KubeadmControlPlane
+apiVersion: controlplane.cluster.x-k8s.io/v1alpha4
```
Comment (CecileRobertMichon) [inline templates/test/ci/patches/apiserver.yaml:2]: @nilo19 your branch is https://github.com/nilo19/cluster-api-provider-azure/tree/chore/mix not rebased (297 commits behind main) this should be v1beta1. Try rebasing and aplying the changes on the newer templates.

Comment (nilo19) [reply templates/test/ci/patches/apiserver.yaml]: Thanks for the info.

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Comment by CecileRobertMichon** [comment]
capz-e2e: Workload cluster creation Creating a cluster that uses the external cloud provider with a 1 control plane nodes and 2 worker nodes

passed

/hold cancel

**Pull request merged**